analyses/cell-type-ewings/cell_type_annotations/SCPCS000490/SCPCL000822_tumor-cell-validation.Rmd

```diff
@@ -0,0 +1,492 @@
+---
+title: "SCPCL000822 - Validation of tumor cells"
+author: Ally Hawkins
+date: "`r Sys.Date()`"
+output:
+  html_document:
+    toc: true
+    toc_depth: 3
+---
+
+## Introduction
+
+In this notebook we compare tumor cell annotations obtained from marker genes, `CopyKAT`, and `InferCNV` to identify a group of tumor cells we are confident in. 
+
+
+- First we look at the cells that are annotated as tumor cells by each method. 
+Using this information, we call any cell that is annotated as a tumor cell by both `CopyKAT` and `InferCNV` as a tumor cell. 
+- We then validate that these are in fact tumor cells by doing the following: 
+  - Looking at marker gene expression. 
+  - Looking at expression of EWS-FLI1 target gene sets. 
+  - Confirming presence of known CNVs in Ewing sarcoma
+
+## Setup
+
+```{r packages}
+suppressPackageStartupMessages({
+  # load required packages
+  library(SingleCellExperiment)
+  library(ggplot2)
+})
+
+# Set default ggplot theme
+theme_set(
+  theme_bw()
+)
+```
+
+
+```{r base paths}
+# The base path for the OpenScPCA repository, found by its (hidden) .git directory
+repository_base <- rprojroot::find_root(rprojroot::is_git_root)
+
+# The current data directory, found within the repository base directory
+data_dir <- file.path(repository_base, "data", "2024-05-01")
+sample_dir <- file.path(data_dir, "SCPCP000015", "SCPCS000490")
+
+# The path to this module
+module_base <- file.path(repository_base, "analyses", "cell-type-ewings")
+```
+
+```{r}
+# Input files
+sce_file <- file.path(sample_dir, "SCPCL000822_processed.rds")
+marker_genes <- file.path(module_base, "references", "tumor-marker-genes.tsv")
+
+# results from annotation workflow 
+results_dir <- file.path(module_base, "results", "annotate_tumor_cells_output", "SCPCS000490")
+
+marker_gene_results_file <- file.path(results_dir, "SCPCL000822_tumor-normal-classifications.tsv")
+copykat_predictions_file <- file.path(results_dir, "SCPCL000822_copykat-classifications.tsv")
+
+infercnv_predictions_file <- file.path(results_dir, "SCPCL000822_infercnv-classifications.tsv")
+infercnv_metadata_file <- file.path(results_dir, "infercnv", "SCPCL000822_cnv-metadata.tsv")
+
+geneset_scores_file <- file.path(results_dir, "SCPCL000822_gene-set-scores.tsv")
+
+# output files 
+final_annotations_dir <- file.path(module_base, "cell_type_annotations", "SCPCS000490")
+fs::dir_create(final_annotations_dir)
+final_annotations_file <- file.path(final_annotations_dir, "SCPCL000822_tumor-classifications.tsv")
+```
+
+
+```{r}
+# read in all files 
+sce <- readr::read_rds(sce_file)
+
+# read in marker genes table 
+marker_genes_df <- readr::read_tsv(marker_genes) |> 
```

Comment:
A minor comment, but do we maybe want to throw in `show_col_types = FALSE`  or `col_types = cols(.default = "c"))` just to quiet the `readr` messages.

Or do it once with `options(readr.show_col_types = FALSE)` in setup?